File: MaWPhotos/src/main/java/us/mikeandwan/photos/services/AuthInterceptor.java
package us.mikeandwan.photos.services;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;


public class AuthInterceptor implements Interceptor {
    private AuthStateManager _authStateManager;

    public AuthInterceptor(AuthStateManager authStateManager) {
        _authStateManager = authStateManager;
    }


    @Override
    public synchronized Response intercept(Interceptor.Chain chain) throws IOException {
        Request srcRequest = chain.request();

        Request request = srcRequest.newBuilder()
            .addHeader("Authorization", String.format("Bearer %s", _authStateManager.getCurrent().getAccessToken()))
            .build();

        return chain.proceed(request);
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/services/AuthInterceptor.kt
package us.mikeandwan.photos.services

import us.mikeandwan.photos.services.AuthStateManager
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import kotlin.jvm.Synchronized
import kotlin.Throws

class AuthInterceptor(private val _authStateManager: AuthStateManager) : Interceptor {
    @Synchronized
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val srcRequest = chain.request()
        val request = srcRequest.newBuilder()
            .addHeader(
                "Authorization",
                String.format("Bearer %s", _authStateManager.current.accessToken)
            )
            .build()
        return chain.proceed(request)
    }
}